examples/threshold_signature/registration.rs

```diff
@@ -0,0 +1,106 @@
+use blake2::digest::{Update, VariableOutput};
+use blake2::Blake2bVar;
+use blst::min_sig::PublicKey;
+use std::collections::HashMap;
+
+/// A hash table for storing BLS public keys with their indices
+#[derive(Debug, Clone)]
+pub(crate) struct RegisteredKeys {
```

Comment:
I am not sure you actually need the signer index, but only being able to know they are in the struct. Perhaps using a [hashset](https://doc.rust-lang.org/std/collections/hash_set/struct.HashSet.html) instead would be more efficient?